examples/rust/bzsstack/src/syscall.rs

```diff
@@ -0,0 +1,81 @@
+extern crate libc;
+
+#[repr(C)]
+pub union sample_un {
+    pub sample_period: u64,
+    pub sample_freq: u64,
+}
+
+#[repr(C)]
+pub union wakeup_un {
+    pub wakeup_events: u32,
+    pub wakeup_atermark: u32,
+}
+
+#[repr(C)]
+pub union bp_1_un {
+    pub bp_addr: u64,
+    pub kprobe_func: u64,
+    pub uprobe_path: u64,
+    pub config1: u64,
+}
+
+#[repr(C)]
+pub union bp_2_un {
+    pub bp_len: u64,
+    pub kprobe_addr: u64,
+    pub probe_offset: u64,
+    pub config2: u64,
+}
+
+#[repr(C)]
+pub struct perf_event_attr {
```

Comment:
btw, latest libbpf-sys package (which libbpf-rs is also using) should have all the perf_event definitions, so you should be able to reuse it